packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,286 @@
+    />
+  );
+}) as ToolbarButtonPrimitive;
+
+ToolbarButton.displayName = BUTTON_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarLink
+ * -----------------------------------------------------------------------------------------------*/
+
+const LINK_NAME = 'ToolbarLink';
+const LINK_DEFAULT_TAG = 'a';
+
+type ToolbarLinkOwnProps = Polymorphic.OwnProps<typeof ToolbarButton>;
+type ToolbarLinkPrimitive = Polymorphic.ForwardRefComponent<
+  typeof LINK_DEFAULT_TAG,
+  ToolbarLinkOwnProps
+>;
+
+const ToolbarLink = React.forwardRef((props, forwardedRef) => {
+  const { as = LINK_DEFAULT_TAG, ...linkProps } = props;
+  return (
+    <ToolbarButton
+      {...linkProps}
+      as={as}
+      selector={getSelector(LINK_NAME)}
+      ref={forwardedRef}
+      onKeyDown={composeEventHandlers(linkProps.onKeyDown, (event) => {
+        if (event.key === ' ') {
+          event.currentTarget.click();
+        }
+      })}
+    />
+  );
+}) as ToolbarLinkPrimitive;
+
+ToolbarLink.displayName = LINK_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarRadioGroup
+ * -----------------------------------------------------------------------------------------------*/
+
+const RADIO_GROUP_NAME = 'ToolbarRadioGroup';
+
+type ToolbarRadioGroupOwnProps = Polymorphic.OwnProps<typeof RadioGroupPrimitive.Root>;
+type ToolbarRadioGroupPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof RadioGroupPrimitive.Root>,
+  ToolbarRadioGroupOwnProps
+>;
+
+const ToolbarRadioGroup = React.forwardRef((props, forwardedRef) => {
+  const context = useToolbarContext(RADIO_GROUP_NAME);
+  return (
+    <RadioGroupPrimitive.Root
+      data-orientation={context.orientation}
+      {...props}
+      selector={getSelector(RADIO_GROUP_NAME)}
+      ref={forwardedRef}
+      rovingFocus={false}
+    />
+  );
+}) as ToolbarRadioGroupPrimitive;
+
+ToolbarRadioGroup.displayName = RADIO_GROUP_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarRadioItem
+ * -----------------------------------------------------------------------------------------------*/
+
+const RADIO_GROUP_ITEM_NAME = 'ToolbarRadioItem';
+
+type ToolbarRadioOwnProps = Merge<
+  Polymorphic.OwnProps<typeof RadioGroupPrimitive.Item>,
+  Polymorphic.OwnProps<typeof ToolbarButton>
+>;
+type ToolbarRadioPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof ToolbarButton>,
+  ToolbarRadioOwnProps
+>;
+
+const ToolbarRadioItem = React.forwardRef((props, forwardedRef) => (
+  <RadioGroupPrimitive.Item
+    {...props}
+    as={ToolbarButton}
+    selector={getSelector(RADIO_GROUP_ITEM_NAME)}
+    ref={forwardedRef}
+  />
+)) as ToolbarRadioPrimitive;
+
+ToolbarRadioItem.displayName = RADIO_GROUP_ITEM_NAME;
```

Comment:
If we're not doing anything special, should we use the `extendComponent` util?

Reply:
I am doing `as={ToolbarButton}` here and merging types which is why I needed to expand it like this

Reply:
Oh I missed that. Does that mean that `as` is taken then and can't be used on `ToolbarRadioItem`?

Reply:
Ah, good point, let me fix that ...

Reply:
Fixed :+1: good spot 👀